lib/extractor/layer.ts

```diff
@@ -53,6 +55,73 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
+  ociArchiveFilesystemPath: string,
+  extractActions: ExtractAction[],
+): Promise<{
+  layers: ExtractedLayers[];
+  manifest: OciArchiveManifest;
+}> {
+  return new Promise((resolve, reject) => {
+    const tarExtractor: Extract = extract();
+
+    const layers: Record<string, ExtractedLayers> = {};
+    const manifests: Record<string, OciArchiveManifest> = {};
+    let imageIndex;
```

Comment:
This is a bit misleading because it has type `any`, it should be `OciImageIndex | undefined` and you will find that there is a potential undefined check that we don't handle! And it can result in an error where we call `imageIndex.manifests.find()`

Actually according to the spec the index.json is _optional_ https://github.com/opencontainers/image-spec/blob/master/image-index.md

Also the `manifests[]` may be of size 0. Maybe it's a bit too early to handle these cases but they are possible according to the spec and we should think how to act when it happens!

Reply:
I think I definitely need to handle image index being optional at this point though, wdyt? It is optional, so not doing it would be just not handling a case. 
In case of manifests array being size of 0, I'm wondering if it would be enough to handle it when we see that case, possibly add some logging here.

Reply:
I guess a manifest.length of 0 should mean we return no analysis? But if index.json is missing then probably we can't process the image either so we could throw a user error? Not sure, could check with Gareth as well

Reply:
I posted on slack and checked with Gareth :-)